### PR TITLE
Added Makefile and hooked up Circle to the client build

### PR DESCRIPTION
**Pull request by tenshiemi**

### CODE REVIEW

File: bin/install-dependencies.sh [hunk 1]
```diff
@@ -2,6 +2,18 @@
...
 
+# Install utilities
+
+# Throwing away stdout logs because they were too plentiful and
+# Circle couldn't display them in their web interface.
+# Errors should still post to the console.
+apt-get update
+apt-get -qq install -y git-all build-essential curl > /dev/null
+
+# Install brew
```
Comment (brainsik) [inline bin/install-dependencies.sh]: You definitely don't want to do this on a Linux system. Homebrew is for macOS. We should rely on the the Linux packaging system to install any dependencies.

File: bin/install-dependencies.sh [hunk 2]
```diff
@@ -2,6 +2,18 @@
...
+/usr/bin/ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"
+brew install shellcheck
```
Comment (brainsik) [inline bin/install-dependencies.sh]: ShellCheck already exists on our containers. Here's the list of tools we add: https://github.com/trussworks/circleci-docker-primary#readme

File: bin/install-dependencies.sh (hunk 1 above)
Comment (tenshiemi) [reply bin/install-dependencies.sh]: Fixed!

File: bin/install-dependencies.sh (hunk 2 above)
Comment (tenshiemi) [reply bin/install-dependencies.sh]: What about build-essential and curl?

Comment (brainsik) [reply bin/install-dependencies.sh]: I'm not sure if they are on the base image or not. You could launch it in a local docker and inspect. Or just try installing them and see what the output is.

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+# This target ensures that the pre-commit hook is installed and kept up to date
+# if pre-commit updates.
+pre-commit: .git/hooks/pre-commit
+.git/hooks/pre-commit: /usr/local/bin/pre-commit
+	pre-commit install
+
+client_deps:
+	cd client && \
+	yarn --frozen-lockfile install
```
Comment (ntwyman) [inline Makefile]: Is the frozen-lockfile needed given the .yarnrc below?

File: client/yarn.rc [hunk 4]
```diff
@@ -0,0 +1 @@
+install.frozen-lockfile true
```
Comment (ntwyman) [inline client/yarn.rc]: I think this file should be called `.yarn.rc`?

File: client/yarn.lock [hunk 5]
```diff
@@ -5119,7 +5119,7 @@ react-dev-utils@^4.2.1:
     strip-ansi "3.0.1"
     text-table "0.2.0"
 
-react-dom@16.2.0:
+react-dom@^16.2.0:
```
Comment (ntwyman) [inline client/yarn.lock]: Did you mean for this to get added? I though you were looking to pin versions.

File: .circleci/config.yml [hunk 6]
```diff
@@ -3,6 +3,9 @@ jobs:
   build:
     docker:
       - image: trussworks/circleci-docker-primary:49e04ed364985ca83145d9336b0cb82ba599ebe0
+    environment:
+      - GOPATH: "/root/ppp/server"
```
Comment (ntwyman) [inline .circleci/config.yml]: Does this exist yet?

**Pull request merged**